src/main.rs

```diff
@@ -135,30 +135,21 @@ fn setup(args: &[String]) -> Setup {
     let mixer = mixer::find(mixer_name.as_ref())
         .expect("Invalid mixer");
 
-    let initial_volume: i32;
-    if matches.opt_present("initial-volume") && matches.opt_str("initial-volume").unwrap().parse::<i32>().is_ok() {
-        let iv = matches.opt_str("initial-volume").unwrap().parse::<i32>().unwrap();
-        match iv {
-            iv if iv >= 0 && iv <= 100 => { initial_volume = iv * 0xFFFF / 100 }
-            _ => {
-                debug!("Volume needs to be a value from 0-100; set volume level to 50%");
-                initial_volume = 0x8000;
+    let initial_volume = matches
+        .opt_str("initial-volume")
+        .map(|volume| {
+            let volume = volume.parse::<i32>().unwrap();
+            if volume < 0 || volume > 100 {
+                panic!("Initial volume must be in the range 0-100");
             }
-        }
-    } else {
-        initial_volume = 0x8000;
-    }
-
-    let zeroconf_port: u16;
-    if matches.opt_present("zeroconf-port") && matches.opt_str("zeroconf-port").unwrap().parse::<u16>().is_ok() {
-        let z = matches.opt_str("zeroconf-port").unwrap().parse::<u16>().unwrap();
-        match z {
-            z if z >= 1024 => { zeroconf_port = z }
-            _ => { zeroconf_port = 0 }
-        }
-    } else {
-        zeroconf_port = 0
-    }
+            volume * 0xFFFF / 100
+        })
+        .unwrap_or(0x8000);
+
+    let zeroconf_port =
+    matches.opt_str("zeroconf-port")
+           .map(|volume| volume.parse::<u16>().unwrap())
```

Comment:
Please correct me if I'm wrong, but this shouldn't be volume should it?

Reply:
nope, it shouldn't. Probably a copypasta. Also, we need to handle the error when a non-root user tries to set a priviledged port (<1024). Currently it panics with a cryptic error, we need to display a proper error, or default to port 0.

Reply:
I would be more inclined to say it should through an error (that a human can understand) and quit, as more that likely if you're changing this port its for a reason. However talking of panics like you said elsewhere I don't think a volume out of range is panic worthy, just set as default and show a debug message.

Reply:
I think we need to make a decision now with regards to how we approach problems like this in future. Do we want librespot to be user friendly, or require precision? If the latter, we should just fail hard on any bad input, or we accomodate bad inputs and throw warnings. Either way, we need to be consistent...

Reply:
As we're planning a daemon for the future that is going to be user friendly, it can be responsible for potentially fixing inputs. Librespot on its own should require precision as it is after all a library.
Thoughts?

Reply:
I agree. So it should panic on initial-volume as paul suggested, and also on priviledged ports when not run as root.